prow/config/config_test.go

```diff
@@ -1156,3 +1156,34 @@ func TestDecorationDefaulting(t *testing.T) {
 		}
 	}
 }
+
+func TestBranchProtection(t *testing.T) {
+	cases := []struct {
+		org    string
+		repo   string
+		branch string
+	}{
+		{
+			org:    "kubernetes",
+			repo:   "test-infra",
+			branch: "all-branches",
+		},
+		{
+			org:    "kubernetes-sigs",
+			repo:   "any-repo",
+			branch: "any-branch",
+		},
+	}
+	for _, tc := range cases {
+		policy, err := c.GetBranchProtection(tc.org, tc.repo, tc.branch)
+		if err != nil {
+			t.Errorf("%s/%s=%s: unexpected error: %v", tc.org, tc.repo, tc.branch, err)
+		}
+		if policy.Protect == nil || !*policy.Protect {
+			t.Errorf("%s/%s=%s: unprotected: %v", tc.org, tc.repo, tc.branch, policy.Protect)
+		}
+		if !sets.NewString(policy.RequiredStatusChecks.Contexts...).Has("cla/linuxfoundation") {
```

Comment:
it seems a little off to couple tests to the kubernetes/* prow config. /shrug

Reply:
I mean we have plenty of that now, but it seems like a direction we should move away from..?

Reply:
I agree we should separate kubernetes config testing from prow/config package testing... But I think config_test.go already conflates this, and I would rather pull these apart in its own PR

Reply:
SGTM